java/dynamodb-tools/src/test/java/sleeper/dynamodb/tools/DynamoDBUtilsPagingIT.java

```diff
@@ -218,7 +255,47 @@ void shouldReturnPagedResultsWhenRecordsEqualToScanLimit() {
         }
 
         @Test
-        void shouldReturnPagedResultsWhenLastPageHasFewerRecordsThanScanLimit() {
+        void shouldNotReturnEmptyResultWhenFewerRecordsThanLimit() {
+            // Given
+            Map<String, AttributeValue> record1 = new DynamoDBRecordBuilder()
+                    .string(TEST_KEY, "test-key")
+                    .string(TEST_VALUE, "test-value-1").build();
+            Map<String, AttributeValue> record2 = new DynamoDBRecordBuilder()
+                    .string(TEST_KEY, "test-key")
+                    .string(TEST_VALUE, "test-value-2").build();
+
+            dynamoDBClient.putItem(new PutItemRequest(tableName, record1));
+            dynamoDBClient.putItem(new PutItemRequest(tableName, record2));
+
+            // When/Then
+            assertThat(streamPagedResults(dynamoDBClient, new ScanRequest()
```

Comment:
Shouldn't this be using a query instead of a scan?

Reply:
Good catch, thanks. Fixed.